iceoryx_posh/test/moduletests/test_roudi_portpool.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2021 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_posh/internal/roudi/port_pool_data.hpp"
+#include "iceoryx_posh/internal/runtime/node_data.hpp"
+#include "iceoryx_posh/roudi/port_pool.hpp"
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+using namespace ::testing;
+using ::testing::Return;
+
+namespace example
+{
+namespace test
+{
+class PortPool_test : public Test
+{
+  public:
+    iox::roudi::PortPoolData* portPoolData = new iox::roudi::PortPoolData{};
+};
+
+TEST_F(PortPool_test, AddNodeDataFailsWhenNodeListIsFull)
+{
+    auto errorHandlerCalled{false};
+    iox::Error errorHandlerType;
+    auto errorHandlerGuard = iox::ErrorHandler::SetTemporaryErrorHandler(
+        [&](const iox::Error error, const std::function<void()>, const iox::ErrorLevel) {
+            errorHandlerType = error;
+            errorHandlerCalled = true;
+        });
+
+    for (uint32_t i = 0; i < iox::MAX_NODE_NUMBER; ++i)
+    {
+        portPoolData->m_nodeMembers.insert("processName", "nodeName", i);
```

Comment:
please use `addNodeData` function like below

Reply:
m_nodeMembers is iox::roudi::FixedPositionContainer which doest not have addNodeData function

Reply:
Yes i see that, but why are you inserting the nodename directly into the container instead of using `addNodeData` from the Portpool?
See comment: https://github.com/eclipse-iceoryx/iceoryx/pull/516#discussion_r562094828

Reply:
done